### PR TITLE
[cli] Fix `ti test`

### PR DESCRIPTION
**Pull request by yuanming-hu**
Related issue = #773 

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

### CODE REVIEW

File: python/taichi/lang/__init__.py [hunk 1]
```diff
@@ -251,6 +251,8 @@ def supported_archs():
     if ti.core.with_opengl():
         archs.append(opengl)
     wanted_archs = os.environ.get('TI_WANTED_ARCHS', '').split(',')
+    # Note, ''.split(',') gives you [''], which is not an empty array.
+    wanted_archs = list(filter(lambda x: x != '', wanted_archs))
```
Comment (archibate) [inline python/taichi/lang/__init__.py:255]: Nice catch!,maybe simply len instead of lambda x: x != '' works?

Comment (yuanming-hu) [reply python/taichi/lang/__init__.py:255]: I think both will work and `x != ''` seems clearer to me. Thanks for the prompt review - I'll merge once CI passes.

File: python/taichi/lang/__init__.py [hunk 2]
```diff
@@ -251,6 +251,8 @@ def supported_archs():
...
     if len(wanted_archs):
         archs, old_archs = [], archs
         for arch in old_archs:
```
Comment (archibate) [inline python/taichi/lang/__init__.py:258]: May also want arch_from_name?

Comment (yuanming-hu) [reply python/taichi/lang/__init__.py:258]: Maybe, but let me quickly merge this in after CI passes. Feel free to do that in another PR. Thanks!

**Pull request merged**